source/JsMaterialX/JsMaterialXCore/JsMaterial.cpp

```diff
@@ -0,0 +1,45 @@
+#include "../vectorHelper.h"
+#include "../helpers.h"
+
+#include <MaterialXCore/Material.h>
+
+#include <emscripten/bind.h>
+
+#include <unordered_set>
+#include <vector>
+
+namespace ems = emscripten;
+namespace mx = MaterialX;
+
+extern "C"
+{
+    EMSCRIPTEN_BINDINGS(material)
+    {
+        ems::function("getShaderNodes", ems::optional_override([](mx::NodePtr materialNode) {
+            std::unordered_set<mx::NodePtr> set =  mx::getShaderNodes(materialNode);
+            // Put all elements into an std::vector
+            std::vector<mx::NodePtr> vec;
+            vec.insert(vec.end(), set.begin(), set.end());
+            return vec;
```

Comment:
This could be done with the `arrayToVec` helper. Note: Don't try to construct a `val::array` directly here, as it will likely not work with smart pointers. Same goes for the other bindings in this file.